geospatial/src/Dem.cc

```diff
@@ -110,21 +134,32 @@ int Dem::Load(const std::string &_filename)
 
   this->dataPtr->filename = fullName;
 
-  if (!exists(findFilePath(fullName)))
+  // Create a re-usable lambda for opening a dataset.
+  auto OpenInGdal = [this](const std::string& name) -> bool
   {
-    gzerr << "Unable to find DEM file[" << _filename << "]." << std::endl;
-    return -1;
-  }
+    GDALDatasetH handle = GDALOpen(name.c_str(), GA_ReadOnly);
+    if (handle) {
+      this->dataPtr->dataSet = GDALDataset::FromHandle(handle);
```

Comment:
Glad we can get rid of the `reinterpret_cast`. Can you do the same in the destructor?

Reply:
https://gdal.org/api/raster_c_api.html#_CPPv49GDALClose12GDALDatasetH
Yep! Added in 5e03fc3488c1416266e57db9b5e7bd7008eeec83.